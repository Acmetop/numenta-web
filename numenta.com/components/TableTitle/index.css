.border {
  composes: b--black-10 from 'tachyons';
  composes: ba from 'tachyons';
}

.tableTitle {
  composes: bg-near-white from 'tachyons';
  composes: dib from 'tachyons';
  composes: dtc-ns from 'tachyons';
  composes: pa2 from 'tachyons';
  composes: w-100 from 'tachyons';
  composes: w-auto-ns from 'tachyons';
}
