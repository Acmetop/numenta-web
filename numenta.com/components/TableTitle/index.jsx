import React from 'react'

import styles from './index.css'


const TableTitle = ({children, colSpan}) => (
  <th className={styles.tableTitle} colSpan={colSpan}>
    {children}
  </th>
)

TableTitle.propTypes = {
  children: React.PropTypes.any,
  colSpan: React.PropTypes.number.isRequired,
}

TableTitle.defaultProps = {
  colSpan: 1,
}

export default TableTitle
