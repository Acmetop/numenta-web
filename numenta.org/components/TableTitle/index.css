/**
 * Numenta.org HTM Community website source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

.border {
  composes: b--black-10 from 'tachyons';
  composes: ba from 'tachyons';
}

.tableTitle {
  composes: bg-near-white from 'tachyons';
  composes: pa2 from 'tachyons';
  composes: v-top from 'tachyons';
}
