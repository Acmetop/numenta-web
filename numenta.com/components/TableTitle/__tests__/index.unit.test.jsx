import React from 'react'
import renderer from 'react-test-renderer'

import TableTitle from '../../TableTitle'


describe('TableTitle React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <TableTitle>
        Some Statistics
      </TableTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
