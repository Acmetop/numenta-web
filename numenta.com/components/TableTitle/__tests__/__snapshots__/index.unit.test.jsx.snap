exports[`TableTitle React component Renders correctly 1`] = `
<th
  className="tableTitle"
  colSpan={1}>
  Some Statistics
</th>
`;
